src/main/java/mServer/crawler/sender/br/BrFilmDeserializer.java

```diff
@@ -0,0 +1,393 @@
+  private static final String JSON_ELEMENT_PUBLIC_LOCATION = "publicLocation";
+
+  private static final String JSON_ELEMENT_VIDEO_PROFILE = "videoProfile";
+  private static final String JSON_ELEMENT_WIDTH = "width";
+
+  private final MediathekReader crawler;
+  private final String filmId;
+  
+  private final DateTimeFormatter dateFormatDatenFilm = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+  private final DateTimeFormatter timeFormatDatenFilm = DateTimeFormatter.ofPattern("HH:mm:ss");
+
+  public BrFilmDeserializer(final MediathekReader aCrawler, final String aFilmId) {
+    crawler = aCrawler;
+    filmId = aFilmId;
+  }
+
+  /**
+   * Resolves the Film details which and creates a Film of it.<br>
+   * The data has this structure:
+   * <code>data -> viewer -> clip -> videoFiles -> edges[] -> node -> id</code><br>
+   * <code>data -> viewer -> detailClip -> title</code><br>
+   * <code>data -> viewer -> detailClip -> kicker</code><br>
+   * <code>data -> viewer -> detailClip -> duration</code><br>
+   * <code>data -> viewer -> detailClip -> broadcasts -> edges[0] -> node -> start</code><br>
+   * Optional: <code>data -> viewer -> detailClip -> shortDescription</code><br>
+   * Optional: <code>data -> viewer -> detailClip -> description</code>
+   */
+  @Override
+  public Optional<DatenFilm> deserialize(final JsonElement aElement, final Type aType,
+      final JsonDeserializationContext aContext) {
+    try {
+      final Optional<JsonObject> viewer = getViewer(aElement.getAsJsonObject());
+      if (viewer.isPresent()) {
+
+        final Optional<JsonObject> detailClip = getDetailClip(viewer.get());
+
+        return buildFilm(detailClip, viewer.get());
+
+
+      } else {
+        printMissingDetails(JSON_ELEMENT_VIEWER);
+      }
+    } catch (final UnsupportedOperationException unsupportedOperationException) {
+      // This will happen when a element is JsonNull.
+      LOG.error("BR: A needed JSON element is JsonNull.", unsupportedOperationException);
+      FilmeSuchen.listeSenderLaufen.inc(crawler.getSendername(), RunSender.Count.FEHLER);
+    }
+
+    return Optional.empty();
+  }
+
+  private String getDescriptions(final JsonObject aDetailClip) {
+    String description = "";
+    
+    if (aDetailClip.has(JSON_ELEMENT_DESCRIPTION)
+        && !aDetailClip.get(JSON_ELEMENT_DESCRIPTION).isJsonNull()) {
+      description = aDetailClip.get(JSON_ELEMENT_DESCRIPTION).getAsString();
+    } else if (aDetailClip.has(JSON_ELEMENT_SHORT_DESCRIPTION)
+        && !aDetailClip.get(JSON_ELEMENT_SHORT_DESCRIPTION).isJsonNull()) {
+      description = aDetailClip.get(JSON_ELEMENT_SHORT_DESCRIPTION).getAsString();
+    }
+    
+    return description;
+  }
+
+  private Map<Resolution, String> getUrls(final JsonObject viewer) {
+    Map<Resolution, String> urlMap = new HashMap<>();
+    
+    final Set<BrUrlDTO> urls = edgesToUrls(viewer);
+      if (!urls.isEmpty()) {
+        // TODO ist das wirklich sinnvoll? 
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)